### PR TITLE
Implemented the mainnet deployment scripts

### PR DESCRIPTION
**Pull request by jalextowle**
# Resolved Issues

Addresses: #976, https://github.com/spearbit-audits/delv-week-review/issues/2

# Description

This PR adds the mainnet deployment scripts for the ElementDAO deployment and addresses the last remaining issues from Saw Mon. 

# Review Checklists

Please check each item **before approving** the pull request. While going
through the checklist, it is recommended to leave comments on items that are
referenced in the checklist to make sure that they are reviewed. If there are
multiple reviewers, copy the checklists into sections titled `## [Reviewer Name]`.
If the PR doesn't touch Solidity, the corresponding checklist can
be removed.

## [[Reviewer Name]]

- [ ] **Tokens**
    - [ ] Do all `approve` calls use `forceApprove`?
    - [ ] Do all `transfer` calls use `safeTransfer`?
    - [ ] Do all `transferFrom` calls use `msg.sender` as the `from` address?
        - [ ] If not, is the function access restricted to prevent unauthorized
              token spend?
- [ ] **Low-level calls (`call`, `delegatecall`, `staticcall`, `transfer`, `send`)**
    - [ ] Is the returned `success` boolean checked to handle failed calls?
    - [ ] If using `delegatecall`, are there strict access controls on the
          addresses that can be called? It shouldn't be possible to `delegatecall`
          arbitrary addresses, so the list of possible targets should either be
          immutable or tightly controlled by an admin.
- [ ] **Reentrancy**
    - [ ] Are functions that make external calls or transfer ether marked as `nonReentrant`?
        - [ ] If not, is there documentation that explains why reentrancy is
              not a concern or how it's mitigated?
- [ ] **Gas Optimizations**
    - [ ] Is the logic as simple as possible?
    - [ ] Are the storage values that are used repeatedly cached in stack or
          memory variables?
    - [ ] If loops are used, are there guards in place to avoid out-of-gas
          issues?
- [ ] **Visibility**
    - [ ] Are all `payable` functions restricted to avoid stuck ether?
- [ ] **Math**
    - [ ] Is all of the arithmetic checked or guarded by if-statements that will
          catch underflows?
    - [ ] If `Safe` functions are altered, are potential underflows and
          overflows caught so that a failure flag can be thrown?
    - [ ] Are all of the rounding directions clearly documented?
- [ ] **Testing**
    - [ ] Are there new or updated unit or integration tests?
    - [ ] Do the tests cover the happy paths?
    - [ ] Do the tests cover the unhappy paths?
    - [ ] Are there an adequate number of fuzz tests to ensure that we are
          covering the full input space?

### CODE REVIEW

File: tasks/deploy/hyperdrive.ts [hunk 1]
```diff
@@ -23,7 +23,8 @@ HyperdriveDeployBaseTask(
 
         // deploy the registry
         await run("deploy:registry", {
-            name: `${network.name.toUpperCase()}_REGISTRY`,
+            // TODO: Generalize this.
+            name: "ElementDAO Hyperdrive Registry",
```
Comment (jrhea) [inline tasks/deploy/hyperdrive.ts:27]: why elementdao? this is our registry?  why not delv Hyperdrive registry?

File: tasks/deploy/lib/constants.ts [hunk 2]
```diff
@@ -23,3 +26,5 @@ export const MAINNET_STETH_WHALE =
 
 export const MAINNET_RETH_WHALE =
     "0x1bee69b7dfffa4e2d53c2a2df135c388ad25dcd2" as Address;
+
+export const SIX_MONTHS = "182 days";
```
Comment (jrhea) [inline tasks/deploy/lib/constants.ts:30]: why not 525,600/2 minutes?

Comment (jrhea) [reply tasks/deploy/lib/constants.ts:30]: ed3e1892-63bf-4a63-8028-2a5cc4543613_text

File: tasks/deploy/hyperdrive.ts (hunk 1 above)
Comment (jalextowle) [reply tasks/deploy/hyperdrive.ts:27]: Yeah, that's fair. I'll change it.

**Review by mcclurejt — APPROVED** [review]

File: tasks/deploy/lib/constants.ts (hunk 2 above)
Comment (jalextowle) [reply tasks/deploy/lib/constants.ts:30]: It's not divisible by a day

**Review by jrhea — APPROVED** [review]

**Pull request merged**